examples/05_ensemble_workflow_extend.py

```diff
@@ -97,20 +98,21 @@ def __call__(
         x: torch.Tensor,
         coords: CoordSystem,
     ) -> tuple[torch.Tensor, CoordSystem]:
-        # Construct perturbation
-        dx, coords = self.pm(x, coords)
-        # Find variable in data
+        # Apply perturbation on variable
+        x0, coords0 = map_coords(x, coords, {"variable": np.array(self.variable)})
+        x0, _ = self.pm(x0, coords0)
+        # Add perturbed slice back into original tensor
         ind = np.in1d(coords["variable"], self.variable)
-        dx[..., ~ind, :, :] = 0.0
-        return dx, coords
+        x[..., ind, :, :] = x0
+        return x, coords
```

Comment:
Consider that this solution will not work for the bred vector method, which would require passing the full tensor to the perturbation method.

Reply:
updated to just apply perturbation to whole input, thanks!